File: frontend/src/types/module.ts
import type { LucideIcon } from 'lucide-react'

export type ModuleCategory = 'estrategico' | 'operacional' | 'financeiro' | 'suporte'

export interface ModuleDefinition {
  key: string
  path: string
  name: string
  description: string
  category: ModuleCategory
  icon: LucideIcon
  tags?: string[]
}

File: backend/app/schemas/module.py
from typing import Literal

from app.schemas.base import ORMModel


class ModuleRead(ORMModel):
    key: str
    name: str
    path: str
    category: Literal['estrategico', 'operacional', 'financeiro', 'suporte']
    description: str
    tags: list[str] | tuple[str, ...] = ()



